docs/core/option.md

```diff
@@ -89,19 +159,41 @@ Option<string> greeting =
     select $"{a} {b}";
 ```
 
-## OkOr / OkOrElse
+### Do
 
-You can turn `Option`s into other types, such as `Result`:
+Use `Do` to execute an imperative operation when the option has a value.
 
 ```csharp
-Result<int, string> result = OnlyNonWhitespace("  hello  ")
-    .Map(x => x.Trim())
-    .Map(x => x.Length)
-    .Where(x => x.Length > 0)
-    .OkOr("That string was only whitespace! Bad!")
+Some("Hello")
+    .Do(x => Console.WriteLine(x));
 ```
 
-## Traverse
+## Aggregation and Collection Operations
+
+Functions that are used to aggregate or collect values from multiple containers.
+
+### Somes
+
+Use the `Somes` to extract the values from a sequence of options.
+
+```csharp
+IEnumerable<int> numbers = ListOf
+    .Many(Some(4), Some(3), None, Some(10), None, None, Some(2))
+    .AsEnumerable()
+    .Somes()
+```
+
+### SomesMap
+
+Use the `SomesMap` as a shortcut to `Somes` + `Map`.
+
+```csharp
+IEnumerable<int> lengths = ListOf
+    .Many(Some("Hello"), Some("world"))
```

Comment:
Would `[]` collection syntax work here? Or would that not compile?

Reply:
It does not

> There is no target type for the collection expression